src/add-text-track-data.js

```diff
@@ -64,21 +64,49 @@ const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
   }
 
   if (metadataArray) {
+    let videoDuration;
+
+    if (isNaN(sourceHandler.mediaSource_.duration) ||
+      sourceHandler.mediaSource_.duration === Infinity ||
+      sourceHandler.mediaSource_.duration === -Infinity) {
```

Comment:
I wonder if it is safe to use `Math.abs(sourceHandler.mediaSource_.duration)` instead of checking for both Infinity and -Infinity.